esp-hal/src/hmac.rs

```diff
@@ -327,7 +327,10 @@ impl<'d> Hmac<'d> {
 
         self.byte_written = self.byte_written.wrapping_add(pad_len);
 
-        assert_eq!(self.byte_written % 64, 64 - core::mem::size_of::<u64>());
+        assert_eq!(
+            self.byte_written.is_multiple_of(64) as usize,
+            64 - core::mem::size_of::<u64>()
+        );
```

Comment:
This doesn't look equivalent to me. The left hand side went from 0..63 to 0..=1, the right hand side is 56. In fact this assert makes no sense at all to me.

Reply:
Good catch